tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/RagListener.java

```diff
@@ -148,19 +149,27 @@ public Publisher<Void> reactiveEvent(Event event) {
 
     private Mono<String> addDocumentToRagContext(MailboxEvents.Added addedEvent, MessageResult messageResult) {
         return asRagLearnableContent(messageResult)
-            .doOnSuccess(text -> LOGGER.debug("RAG Listener successfully processed mailContent ***** \n{}\n *****", new DocumentId(messageResult.getThreadId())))
+            .doOnSuccess(text -> LOGGER.debug("RAG Listener successfully processed mailContent ***** \n{}\n *****", new DocumentId(messageResult.getMessageId())))
             .flatMap(content -> openRagHttpClient
                 .addDocument(
                     partitionFactory.forUsername(addedEvent.getUsername()),
-                    new DocumentId(messageResult.getThreadId()),
+                    new DocumentId(messageResult.getMessageId()),
                     content,
                     computeMetaData(addedEvent, messageResult)));
     }
 
     private Map<String, String> computeMetaData(MailboxEvents.Added addedEvent, MessageResult messageResult) {
         try {
-            return  Map.of("date", DateTimeFormatter.ISO_INSTANT.format(parseMessage(messageResult.getFullContent().getInputStream()).getDate().toInstant()),
-                "doctype", "com.linagora.email");
+            Message mimeMessage = parseMessage(messageResult.getFullContent().getInputStream());
+            String text = new MessageContentExtractor()
+                .extract(mimeMessage)
+                .getTextBody()
+                .orElse("");
+            return  Map.of("subject",mimeMessage.getSubject(),
```

Comment:
```suggestion
            return  Map.of("subject", mimeMessage.getSubject(),
```

surprised the ci build dint complain about this... :)